src/recordlinker/linkage/matchers.py

```diff
@@ -0,0 +1,427 @@
+    """
+    return sum(feature_comparisons) == len(feature_comparisons)
+
+
+def eval_log_odds_cutoff(feature_comparisons: list, **kwargs) -> bool:
+    """
+    Determines whether a given set of feature comparisons matches enough
+    to be the result of a true patient link instead of just random chance.
+    This is represented using previously computed log-odds ratios.
+
+    :param feature_comparisons: A list of floats representing the log-odds
+      score of each field computed on.
+    :return: Whether the feature comparisons score well enough to be
+      considered a match.
+    """
+    if "true_match_threshold" not in kwargs:
+        raise KeyError("Cutoff threshold for true matches must be passed.")
+    return sum(feature_comparisons) >= kwargs["true_match_threshold"]
+
+
+def feature_match_exact(
+    record_i: list,
+    record_j: list,
+    feature_col: str,
+    col_to_idx: dict[str, int],
+    **kwargs: dict,
+) -> bool:
+    """
+    Determines whether a single feature in a given pair of records
+    constitutes an exact match (perfect equality).
+
+    :param record_i: One of the records in the candidate pair to evaluate.
+    :param record_j: The second record in the candidate pair.
+    :param feature_col: The name of the column being evaluated (e.g. "city").
+    :param col_to_idx: A dictionary mapping column names to the numeric index
+      in which they occur in order in the data.
+    :return: A boolean indicating whether the features are an exact match.
+    """
+    idx = col_to_idx[feature_col]
+    return record_i[idx] == record_j[idx]
+
+
+def feature_match_four_char(
+    record_i: list,
+    record_j: list,
+    feature_col: str,
+    col_to_idx: dict[str, int],
+    **kwargs: dict,
+) -> bool:
+    """
+    Determines whether a string feature in a pair of records exactly matches
+    on the first four characters.
+
+    :param record_i: One of the records in the candidate pair to evaluate.
+    :param record_j: The second record in the candidate pair.
+    :param feature_col: The name of the column being evaluated (e.g. "city").
+    :param col_to_idx: A dictionary mapping column names to the numeric index
+      in which they occur in order in the data.
+    :return: A boolean indicating whether the features are a match.
+    """
+    idx = col_to_idx[feature_col]
+    first_four_i = record_i[idx][: min(4, len(record_i[idx]))]
+    first_four_j = record_j[idx][: min(4, len(record_j[idx]))]
+    return first_four_i == first_four_j
+
+
+def feature_match_fuzzy_string(
+    record_i: list,
+    record_j: list,
+    feature_col: str,
+    col_to_idx: dict[str, int],
+    **kwargs: dict,
+) -> bool:
+    """
+    Determines whether two strings in a given pair of records are close
+    enough to constitute a partial match. The exact nature of the match
+    is determined by the specified string comparison function (see
+    harmonization/utils/compare_strings for more details) as well as a
```

Comment:
Is this comment out-of-date now that the RL BB is separated from the Harmonization BB?

Reply:
will change shortly